### DISCUSSION TITLE
Data-Importer imports future transactions

### DISCUSSION THREAD

**Discussion by papa-fuchs** [firefly-iii/firefly-iii#9211] [Firefly III]
Hi 
I have the problem with GoCardless imports via Data-importer that future transactions are also imported. This causes a few problems:
1. transaction details changes (description) (is probably a ING Issue)
2. some future transactions are always postponed to the current day
3. transactions cause duplicates in unfavorable cases

So far I have adjusted the import so that it only imports transactions up to yesterday. Here I can make sure that the transactions are always complete.
Unfortunately, the solution is not very nice on the one hand and on the other hand I would have to re-implement it after a server move to Docker.

It would be best if the option “until yesterday” could be integrated directly into the template. But maybe one of you has a better idea.

**Comment by JC5** [comment]
Hey great question!

Check out: https://docs.firefly-iii.org/references/faq/data-importer/salt-edge-gocardless/#gocardless-imports-weird-transactions

**Reply by papa-fuchs** [reply]
thank you for your reply. i try to understand how this help me to identify the transactions.
I think the Problem is, that the future transactions are already booked and not pending.
There is no special Tag i can use, because that is a normal transaction, just not today.

Perhaps i can use a time rule that deletes all transactions that are > yesterday.
I tried to experiment with the date rule today, but than i have the problem that i can not import the transaction again because its a duplicate from a existing transaction even if the existing transaction is already deleted.

Apparently it's already too late today and I can't think clearly anymore, maybe I'll just go to bed and look at the problem again tomorrow.
